plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -102,13 +108,47 @@ protected String getRefreshedLocation(boolean invalidateCaches)
     protected void commitNewTable(TableMetadata metadata)
     {
         verify(version.isEmpty(), "commitNewTable called on a table which already exists");
+        boolean cleanupOnFailure = true;
         String newMetadataLocation = writeNewMetadata(metadata, 0);
         TableInput tableInput = getTableInput(tableName, owner, ImmutableMap.of(METADATA_LOCATION_PROP, newMetadataLocation));
 
         CreateTableRequest createTableRequest = new CreateTableRequest()
                 .withDatabaseName(database)
                 .withTableInput(tableInput);
-        stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+        try {
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+            cleanupOnFailure = false;
+        }
+        catch (AlreadyExistsException
```

Comment:
Why don't we limit the handling to the situations where there can be thrown `CommitStateUnknownException` ?

Reply:
These are the known exception which will be thrown by `glueClient.createTable`